File: Algorithms/C++/Medium/228_Summary_Ranges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        vector<string> ranges;

        if (nums.empty()) {
            return ranges;
        }

        int i = 0;
        while (i < nums.size()) {
            int begin = nums[i];
            int end = nums[i];

            for (i++; i < nums.size() && nums[i] == nums[i - 1] + 1; i++) {
                end = nums[i];
            }

            string range;
            if (begin == end) {
                range = to_string(begin);
            } else {
                range = to_string(begin) + "->" + to_string(end);
            }
            ranges.push_back(range);
        }

        return ranges;
    }
};

File: Algorithms/Swift/Medium/228_Summary_Ranges.swift
class Solution {
    func summaryRanges(nums: [Int]) -> [String] {
        var ranges = [String]()
        
        if nums.isEmpty {
            return ranges
        }
        
        var i = 0
        while i < nums.count {
            var start = nums[i]
            var end = start
            
            i += 1
            while i < nums.count && nums[i] == nums[i - 1] + 1 {
                end = nums[i]
                i += 1
            }
            
            if end == start {
                ranges.append(String(start))
            } else {
                ranges.append(String(start) + "->" + String(end))
            }
        }
        
        return ranges
    }
}
